src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+
+enum Shape {
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+  Circle = 'circle',
+}
+
 export interface Figure {
+  shape: Shape;
+  color: 'red' | 'green' | 'blue',
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('wrong length specified');
+    }
+
+    if ((a + b) <= c || (b + c) <= a || (c + a) <= b) {
+      throw new Error('wrong length specified');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```

Comment:
Here you can also destructure this to make the code more concise.
 const {a, b, c} = this